queryscript/src/compile/generics.rs

```diff
@@ -201,8 +205,28 @@ impl Generic for SumGeneric {
         })?;
         Ok(())
     }
+
+    fn resolve(&self) -> Result<CRef<MType>> {
```

Comment:
The implementation for this and `External` are exactly the same. Can we refactor it?